contracts/asset/ERC20Factory.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+pragma experimental ABIEncoderV2;
+
+import "../MintableERC20.sol";
+import "../SimpleERC20.sol";
+import '../asset/interfaces/IRegistryHub.sol';
+
+/**
+ * @dev Factory contract to create an ERC20 entity
+ *
+ * This is the entry contract that user start to create their own token.
+ */
+contract ERC20Factory {
+    address public registryHub;
+
+    // address => isMintable
+    mapping (address => bool) public mintableList;
+
+    // all ERC20 token deployed by this factory
+    address[] tokenAddresses;
```

Comment:
Could it be better if we split address list into mintableAddress and nonmintableAddress separately? So that client may not need iterator whole list to find both of them